src/doc/reference.md

```diff
@@ -555,28 +555,45 @@ Users of `rustc` can define new syntax extensions in two ways:
 
 ## Macros
 
-`macro_rules` allows users to define syntax extension in a declarative way.  We
-call such extensions "macros by example" or simply "macros" — to be distinguished
-from the "procedural macros" defined in [compiler plugins][plugin].
+`macro_rules` allows users to define syntax extensions in a declarative way.
+We call such extensions "macros by example" or simply "macros" — to be
+distinguished from the "procedural macros" defined in [compiler plugins][plugin].
 
-Currently, macros can expand to expressions, statements, items, or patterns.
+As shown below, the body of a `macro_rules` macro consists of one or more
+semicolon-separated arms, similar in appearance to a `match` statement. On
+the left side of the `=>` is a matcher describing the syntax that the macro
+accepts, optionally capturing parts of it for use in the transcriber, which
+is on the right side of the `=>`. (Choice of delimiters for the macro body,
+matchers, and transcribers is arbitrary.)
 
-(A `sep_token` is any token other than `*` and `+`. A `non_special_token` is
-any token other than a delimiter or `$`.)
+```rust,ignore
+macro_rules! /* name of macro */ {
+    (/* matcher for arm 1 */) => { /* transcriber for arm 1 /* };
+    (/* matcher for arm 2 */) => { /* transcriber for arm 2 /* }
+}
+```
 
-The macro expander looks up macro invocations by name, and tries each macro
-rule in turn. It transcribes the first successful match. Matching and
+Currently, macros can expand to expressions, statements, items, patterns, or
+types[^typemacros]. This distinction is inferred from the context in which the
+macro is invoked, and carefully written macros may work in multiple contexts.
+
+[^typemacros]: Gated by
+[the `type_macros` feature](https://github.com/rust-lang/rust/issues/27245).
+
+The macro expander looks up macro invocations by name, and tries each
+arm in turn. It transcribes the first successful match. Matching and
 transcription are closely related to each other, and we will describe them
 together.
 
-### Macro By Example
+### Macros By Example
```

Comment:
No, just call it "Macro By Example". All the cases where you've substituted here it's fine to use it that way.